parlai/core/torch_generator_agent.py

```diff
@@ -1611,7 +1735,27 @@ def select_paths(self, logprobs, prior_scores, current_length):
         # get the actual word id from residual of the same division
         tok_ids = best_idxs % voc_size
 
-        return (hyp_ids, tok_ids, best_scores)
+        tok_scores = torch.gather(logprobs, 1, tok_ids[None, :]).view(-1)
+
+        vocab_size = logprobs.size()[1]
+        ranks = logprobs.scatter(
+            1,
+            logprobs.argsort(descending=True),
+            torch.arange(vocab_size)
+            .type(torch.FloatTensor)
+            .to(logprobs.device)[None, :]
+            .repeat((self.beam_size, 1))
+            + 1,
+        )
+        tok_ranks = torch.gather(ranks, 1, tok_ids[None, :]).view(-1)
```

Comment:
i think there is a bug here, this needs to be gathered along the `hyp_id` -- see code above, the scores are flattened and then the best indexes are chosen from that flattened score. then we back out which hypothesis id the indices were taken from.

Reply:
(also see comment below about bug in gathering! may be helpful here)

Reply:
Hm I'm not sure I'm following here. Will ping you and then update this once we chat about it